src/layout/Grid.js

```diff
@@ -137,16 +126,238 @@
         this._scrollBarWidth = this.getScrollbarWidth();
     };
 
+    Grid.prototype.calcGridCoordinates = function () {
+        var xCoords = [];
+        this._offsetX = this._element.node().getBoundingClientRect().left + (this.gutter()/2);
+        for(var i = 0;i<this._colCount;i++){
+            xCoords.push((i * this._colSize) + this._offsetX);
+        }
+        var yCoords = [];
+        this._offsetY = this._element.node().getBoundingClientRect().top + (this.gutter()/2);
+        for(var j = 0;j<this._rowCount;j++){
+            yCoords.push((j * this._rowSize) + this._offsetY);
+        }
+        this._gridCoordinates = {x:xCoords,y:yCoords};
+    };
+    
+    Grid.prototype.findCurrentLocation = function (e) {
```

Comment:
This is ok for now - but can you open an issue to refactor "findCurrentLocation" so that it doesn't use loops.